shared/src/sources/balancer/pool_storage.rs

```diff
@@ -83,45 +145,96 @@ impl RegisteredWeightedPool {
     }
 }
 
+impl RegisteredStablePool {
+    pub async fn new(
+        block_created: u64,
+        pool_address: H160,
+        data_fetcher: &dyn PoolInfoFetching,
+    ) -> Result<RegisteredStablePool> {
+        let pool_data = data_fetcher.get_stable_pool_data(pool_address).await?;
+        Ok(RegisteredStablePool {
+            pool_id: pool_data.pool_id,
+            pool_address,
+            tokens: pool_data.tokens,
+            scaling_exponents: pool_data.scaling_exponents,
+            block_created,
+        })
+    }
+
+    /// Errors expected here are propagated from `get_pool_data`.
+    pub async fn from_event(
+        block_created: u64,
+        creation: PoolCreated,
+        data_fetcher: &dyn PoolInfoFetching,
+    ) -> Result<RegisteredStablePool> {
+        Self::new(block_created, creation.pool_address, data_fetcher).await
+    }
+}
+
 /// PoolStorage represents in-memory storage of all deployed Balancer Pools
 #[derive(Derivative)]
 #[derivative(Debug)]
 pub struct PoolStorage {
     /// Used for O(1) access to all pool_ids for a given token
     pools_by_token: HashMap<H160, HashSet<H256>>,
     /// WeightedPool data for a given PoolId
-    pools: HashMap<H256, RegisteredWeightedPool>,
+    weighted_pools: HashMap<H256, RegisteredWeightedPool>,
+    /// StablePool data for a given PoolId
+    stable_pools: HashMap<H256, RegisteredStablePool>,
+    // TODO - either store weighted and stable in separate or together (but pick one!)
+    /// All static Balancer Pool details for a given PoolId
+    registered_pools: HashMap<H256, RegisteredPool>,
```

Comment:
Still need to pick on here - but at this moment there is no need for the separation. All the code is using the mixed field.